modules/test-utils/src/main/java/org/jboss/wsf/test/JBossWSTest.java

```diff
@@ -162,35 +162,28 @@ public static void executeCommand(String command, OutputStream os, String messag
    private static void executeCommand(List<String> command, OutputStream os, String message, Map<String, String> env) throws IOException
    {
       ProcessBuilder pb = new ProcessBuilder(command);
-      if (env != null)
-      {
-         for (String variable : env.keySet())
-         {
+      if (System.getProperty("os.name").toLowerCase().contains("win")) {
+         pb.environment().put("NOPAUSE", "true");
+      }
+      if (env != null) {
+         for (String variable : env.keySet()) {
             pb.environment().put(variable, env.get(variable));
          }
       }
       Process p = pb.start();
       CopyJob inputStreamJob = new CopyJob(p.getInputStream(), os == null ? System.out : os);
       CopyJob errorStreamJob = new CopyJob(p.getErrorStream(), System.err);
-      // unfortunately the following threads are needed because of Windows behavior
       Thread inputJob = new Thread(inputStreamJob);
       Thread outputJob = new Thread(errorStreamJob);
-      try
-      {  
-         inputJob.start();
-         inputJob.join(COPY_JOB_TIMEOUT);
-         outputJob.start();
-         outputJob.join(COPY_JOB_TIMEOUT);
-         int statusCode = p.waitFor();
-         String fallbackMessage = "Process did exit with status " + statusCode; 
-         assertTrue(message != null ? message : fallbackMessage, statusCode == 0);
-      }
-      catch (InterruptedException ie)
-      {
+      inputJob.start();
+      outputJob.start();
+      try {
+         p.waitFor(PROCESS_TIMEOUT, TimeUnit.SECONDS);
+         String fallbackMessage = "Process did exit with status " + p.exitValue();
```

Comment:
exitValue() javadoc
```
Throws:
IllegalThreadStateException – if the process represented by this Process object has not yet terminated
```

Don't we want to handle that situation differently?
Guard with
```
boolean terminatedInTime = p.waitFor(PROCESS_TIMEOUT, TimeUnit.SECONDS);
if (!terminatedInTime) {
//handle and output useful info ?
fail(...);
}
```

Reply:
@jbliznak   It would be better to check if the process is terminated. I think we can directly add something like : 
```
         boolean exited = p.waitFor(PROCESS_TIMEOUT, TimeUnit.SECONDS);
         assertTrue("Process isn't exited in " + PROCESS_TIMEOUT + " seconds", exited);
```

Reply:
yes, that would serve the same purpose (unless we want to do more than just assert)